collections/max_by_test.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "../testing/asserts.ts";
+import { maxBy } from "./max_by.ts";
+
```

Comment:
You should add tests for

- empty input (the current implementation actually has a bug there)
- `NaN` and `Infinity` behaviour
- no mutation (make sure the originally passed array does not get changed, see other functions in the module)

Reply:
> [collections/maxBy] of empty array

is a test case with an empty array as input. can you provide. can you please provide me the test case input that shows the bug?  @LionC

Reply:
> is a test case with an empty array as input. can you provide. can you please provide me the test case input that shows the bug? @LionC

Yes, if you add this test:

```ts
Deno.test("[collections/maxBy] empty input", () => {
  const emptyArray: Array<{ age: number }> = [];
  const max = maxBy(emptyArray, (it) => it.age);

  assertEquals(max, undefined);
});
```

it will fail.

> there are two test cases with `Infinity` and `NaN` included in the test data. any other usage do you have in mind?

Nevermind, must have missed those, it was late^^ Sorry.